src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,29 @@
+package duke.command;
+
+import duke.gui.Ui;
+import duke.storage.Storage;
+import duke.task.TaskList;
+
+/**
+ * Command class am abstract class with common command methods.
+ *
+ * @author yl-ang
+ */
+public abstract class Command {
+    /**
+     * Executes based on the command type and the objects supplied.
+     *
+     * @param taskLst TaskLst object containing the current tasks.
+     * @param ui Ui object containing the user interface messages and scanner object.
+     * @param storage Storage object containing the methods to load and save.
+     * @return Formatted response from executing command.
+     */
+    public abstract String execute(TaskList taskLst, Ui ui, Storage storage);
+
+    /**
+     * Checks and returns if the command is exit.
+     *
+     * @return Boolean if the command is exit.
+     */
+    public abstract boolean isExit();
```

Comment:
I noticed how all your commands have their individual private isExit boolean variables. Maybe a better way can be to have a protected/private isExit variable in Command class that can be accessed by all inheriting classes. 
Inheriting classes can then set the isExit variable through their constructors or super's constructor. 

Overall, helps with duplicate code. This way, isExit() abstract method here in Command can just be 
```java
public boolean isExit() {
    return this.isExit;
}
```

Reply:
Hi j4ck, great advice, will refactor my code to follow this! 👍